CondFormats/SiPhase2TrackerObjects/src/SiPhase2OuterTrackerLorentzAngle.cc

```diff
@@ -0,0 +1,63 @@
+#include "CondFormats/SiPhase2TrackerObjects/interface/SiPhase2OuterTrackerLorentzAngle.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+bool SiPhase2OuterTrackerLorentzAngle::putLorentzAngle(const uint32_t& detid, float value) {
+  std::unordered_map<unsigned int, float>::const_iterator id = m_LA.find(detid);
+  if (id != m_LA.end()) {
+    edm::LogError("SiPhase2OuterTrackerLorentzAngle") << "SiPhase2OuterTrackerLorentzAngle for DetID " << detid
+                                                      << " is already stored. Skippig this put" << std::endl;
+    return false;
+  } else
+    m_LA[detid] = value;
+  return true;
+}
+
+float SiPhase2OuterTrackerLorentzAngle::getLorentzAngle(const uint32_t& detid) const {
+  std::unordered_map<unsigned int, float>::const_iterator id = m_LA.find(detid);
+  if (id != m_LA.end())
+    return id->second;
+  else {
+    edm::LogError("SiPhase2OuterTrackerLorentzAngle")
+        << "SiPhase2OuterTrackerLorentzAngle for DetID " << detid << " is not stored" << std::endl;
+  }
+  return 0;
+}
+
+void SiPhase2OuterTrackerLorentzAngle::getLorentzAngles_PSP(const TrackerGeometry* trackerGeometry,
```

Comment:
This function could be unified with the next two functions by adding `TrackerGeometry::ModuleType` as an extra argument